libsplinter/src/oauth/profile/openid.rs

```diff
@@ -55,10 +56,33 @@ impl ProfileProvider for OpenIdProfileProvider {
             }
         }
 
-        let user_profile = response
+        let mut user_profile = response
             .json::<OpenIdProfileResponse>()
             .map_err(|_| InternalError::with_message("Received unexpected response body".into()))?;
 
+        // If azure openid is being used for authentication make a call to the
+        // microsoft graph api endpoint with the access token to retrieve the
+        // binary data for the authenticated user's profile photo
+        if self.userinfo_endpoint.contains("graph.microsoft.com") {
+            let picture_response = Client::builder()
+                .build()
+                .map_err(|err| InternalError::from_source(err.into()))?
+                .get("https://graph.microsoft.com/beta/me/photo/$value")
+                .header("Authorization", format!("Bearer {}", access_token))
+                .send()
+                .map_err(|err| InternalError::from_source(err.into()))?;
```

Comment:
I'm not sure we should propagate these failures if the picture can't be downloaded. Perhaps log a warning - it doesn't seem catastrophic enough to give up on the whole profile.

Reply:
Updated this to log a warning and set the picture field of the struct to an empty string if the microsoft graph api call fails. The empty string in the picture field will make the ui display the default profile picture icon.